pkg/rpc/server.go

```diff
@@ -0,0 +1,138 @@
+package smrpc
+
+import (
+	fmt "fmt"
+	"io/ioutil"
+	"path/filepath"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	empty "github.com/golang/protobuf/ptypes/empty"
+	"golang.org/x/net/context"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+	mount "k8s.io/mount-utils"
+
+	iscsiutil "github.com/longhorn/go-iscsi-helper/util"
+
+	"github.com/longhorn/longhorn-share-manager/pkg/crypto"
+	"github.com/longhorn/longhorn-share-manager/pkg/server"
+	"github.com/longhorn/longhorn-share-manager/pkg/util"
+	"github.com/longhorn/longhorn-share-manager/pkg/volume"
+)
+
+type ShareManagerServer struct {
+	manager *server.ShareManager
+}
+
+func NewShareManagerServer(manager *server.ShareManager) *ShareManagerServer {
+	return &ShareManagerServer{
+		manager: manager,
+	}
+}
+
+func (s *ShareManagerServer) FilesystemTrim(ctx context.Context, req *FilesystemTrimRequest) (*empty.Empty, error) {
+	volumeName := s.manager.Volume.Name
+
+	devicePath := filepath.Join(server.DevPath, "longhorn", volumeName)
+	if req.IsEncryptedDevice {
+		devicePath = crypto.VolumeMapper(volumeName)
+	}
+	if !volume.CheckDeviceValid(devicePath) {
+		return &empty.Empty{}, fmt.Errorf("volume %v is not valid", volumeName)
+	}
+
+	mountPoints, err := util.FindMountPoints(devicePath)
+	if err != nil {
+		return &empty.Empty{}, errors.Wrapf(err, "failed to find mount points for volume %v", volumeName)
+	}
```

Comment:
Can we directly trim the export path instead?

Reply:
Yeah. Addressed, thank you.